File: src/test/java/com/dotnomi/fabricdependencyinjection/testclasses/postconstruct/success/PostConstructService.java
package com.dotnomi.fabricdependencyinjection.testclasses.postconstruct.success;

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped;
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct;

@ModScoped
@SuppressWarnings("unused")
public class PostConstructService {
  private boolean initialized;

  @PostConstruct
  private void initialize() {
    this.initialized = true;
  }

  public boolean isInitialized() {
    return this.initialized;
  }
}

File: src/test/java/com/dotnomi/fabricdependencyinjection/kotlin/testclasses/postconstruct/success/PostConstructService.kt
package com.dotnomi.fabricdependencyinjection.kotlin.testclasses.postconstruct.success

import com.dotnomi.fabricdependencyinjection.annotation.ModScoped
import com.dotnomi.fabricdependencyinjection.annotation.PostConstruct

@ModScoped
class PostConstructService {
    var isInitialized: Boolean = false
        private set

    @PostConstruct
    private fun initialize() {
        this.isInitialized = true
    }
}
